bytecode/flags.py

```diff
@@ -0,0 +1,297 @@
+# alias to keep the 'bytecode' variable free
+from enum import IntEnum
+from collections import defaultdict
+
+import bytecode as _bytecode
+
+
+class CoFlags(IntEnum):
+    """Possible values of the co_flags attribute of Code object.
+
+    Notes
+    -----
+    We do not rely on inspect values here as some of them are missing and
+    furthermore would be version dependent.
+
+    """
+    CO_OPTIMIZED             = 0x00001  # noqa
+    CO_NEWLOCALS             = 0x00002  # noqa
+    CO_VARARGS               = 0x00004  # noqa
+    CO_VARKEYWORDS           = 0x00008  # noqa
+    CO_NESTED                = 0x00010  # noqa
+    CO_GENERATOR             = 0x00020  # noqa
+    CO_NOFREE                = 0x00040  # noqa
+    # New in Python 3.5
+    CO_COROUTINE             = 0x00080  # noqa
+    CO_ITERABLE_COROUTINE    = 0x00100  # noqa
+    # New in Python 3.6
+    CO_ASYNC_GENERATOR       = 0x00200  # noqa
+
+    # __future__ flags
+    CO_FUTURE_GENERATOR_STOP = 0x80000  # noqa
+
+
+_CAN_DEDUCE_FROM_CODE = ('optimized', 'generator', 'nofree', 'coroutine')
+
+
+class Flags:
+    """Object representing the flags set for the bytecode.
+
+    Flags can be bools indicating their state, or for some of them None
+    indicating that the value should be inferred from the ConcreteBytecode
+    object passed when converting to an integer. If no bytecode is provided the
+    default values are used.
+
+    The flags whose natural values can be inferred are:
+    - optimized
+    - generator
+    - nofree
+    - coroutine
+
+    """
+
+    def __init__(self, int_or_flags=0):
+
+        self._forced = {}
```

Comment:
I don't understand the purpose of forced. Can you add a short comment here explaining its purpose?